cmake/catkin_package.cmake

```diff
@@ -374,6 +377,15 @@ function(_catkin_package)
       endif()
     endforeach()
 
+    set(PKG_COMPILE_OPTIONS ${${PROJECT_NAME}_COMPILE_OPTIONS})
+    separate_arguments(PKG_COMPILE_OPTIONS)
```

Comment:
Nothing prevents the user from specifying `set(${PROJECT_NAME}_COMPILE_OPTIONS "-Wall -DBla -fomit-frame-pointer)`.
As testing for unique options is crucial, we need to split the string into single options.